packages/hash/api/src/task-execution/index.ts

```diff
@@ -17,18 +24,79 @@ export type TaskExecutor = ReturnType<typeof connectToTaskExecutor> &
 
 /**
  * @param config The connection config for connecting to the Task-Executor
- * @returns a connection object with a run_task method
+ * @returns a connection object with a runTask method
  *
  * Note: a pretty dumb method at the moment, but provides a clean boundary where we can slot in logic for a more complex task-execution mechanism later on
  */
 export const connectToTaskExecutor = (config: Config) => {
   return {
-    run_task: async (route: Tasks) => {
-      return await fetch(`http://${config.host}:${config.port}/${route}`)
-        .then((resp) => resp.json())
-        .catch((error) => {
-          throw new Error(`Failed to execute task ${route}: ${error}`);
+    runTask: async (route: Tasks, body?: any) => {
+      const resp = await fetch(
+        `http://${config.host}:${config.port}/${route}`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(body),
+        },
+      );
+
+      if (resp.ok) {
+        return resp.json();
+      } else {
+        throw new Error(
+          `Task Execution failed for ${route}: ${await resp.text()}`,
+        );
+      }
+    },
+  };
+};
+
+/**
+ * Provides ways to look up EntityIds by name, and to create new ones.
+ *
+ * @param db - The DB Adapter to use for queries
+ * @param user - The User that's making the query
+ * @returns an object with a getExisting, and a createNew, method
+ *
+ * @todo - This is temporary and should be removed when we extend the EntityId type to support these actions properly
```

Comment:
Regarding the Asana task, I did ask if we should block merging this PR to quickly extend the model class, but was informed it might be a little complicated. Our conversations about the graph right now and discovering types will address this, and will require us changing the model classes so I worry that us making a task to specifically support finding by name is wasted effort for now (as it will likely be made redundant by the URI work and structural queries).

Reply:
Makes sense, thanks for confirming